internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -351,6 +392,19 @@ func (m DeploymentResource) Attributes() map[string]*pluginsdk.Schema {
 			Type:     pluginsdk.TypeString,
 			Computed: true,
 		},
+
+		"web_application_firewall_status": {
```

Comment:
if the above block becomes web_application_firewall this could go inside as `web_application_firewall.status.x` ?

Reply:
it could even be web_application_firewall.package_status.x and then web_application_firewall .component_versions

Reply:
I initially wanted to follow this approach of nesting the status inside the web_application_firewall block but the problem I encountered is that the status needs to be an attribute but settings is an argument. I didn't find a good way to have them both in the same block. Do you have any suggestions on how to do that?